src/Formik.tsx

```diff
@@ -384,20 +384,28 @@ export class Formik<Values = FormikValues> extends React.Component<
     value: any,
     shouldValidate: boolean = true
   ) => {
-    if (this.didMount) {
-      // Set form field by name
-      this.setState(
-        prevState => ({
-          ...prevState,
-          values: setIn(prevState.values, field, value),
-        }),
-        () => {
-          if (this.props.validateOnChange && shouldValidate) {
-            this.runValidations(this.state.values);
+    return new Promise((resolve, reject) => {
```

Comment:
The code looks a bit complicated. I think the reason is that nesting is deep. So you should use early return. The following code looks good.

```tsx
return new Promise((resolve, reject) => {
  if(!this.didMount) {
    resolve();
    return;
  }
  
  this.setState(prevState =>({
    ...prevState,
    values: setIn(prevState.values, field, value)
  }),
  () => {
    if(!this.props.validateOnChange || !shouldValidate) {
       resolve();
       return;
    }

    this.runValidations(this.state.values)
      .then(() => resolve())
      .catch(reject);
  })
})
```